### PR TITLE
[@types/vector2djs] add new vector2djs package types.

### PR DESCRIPTION
**Pull request by Leo-Nicolle**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## vector2js ([<kbd>unpkg</kbd>](https://unpkg.com/browse/vector2js@latest/))
was missing the following properties:
1. fromRadians
1. fromDegrees
1. fromString
1. fromArray
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against c0db21fbd544fefd7461298877405b9a44e3a94b
</p>

File: types/vector2js/tslint.json [hunk 1]
```diff
@@ -0,0 +1,2 @@
+{ "extends": "@definitelytyped/dtslint/dt.json"
+}
```
Comment (peterblazejewicz) [inline types/vector2js/tslint.json:2]:
```suggestion
{ "extends": "@definitelytyped/dtslint/dt.json" }
```

File: types/vector2js/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,403 @@
+// Type definitions for vector2js 2.0
+// Project: https://github.com/RonenNess/Vector2js
+// Definitions by: Leo Nicolle <https://github.com/Leo-Nicolle>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+export = Vector2;
+
+type inputVec = Vector2 | { x: number; y: number };
```
Comment (peterblazejewicz) [inline types/vector2js/index.d.ts:7]:
```ts
type InputVec = Vector2 | { x: number; y: number };
```
you would have to rename other references.

File: types/vector2js/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,403 @@
...
+declare class Vector2 {
```
Comment (peterblazejewicz) [inline types/vector2js/index.d.ts:8]:
```suggestion
declare class Vector {
```
the symbol is named `Vector`

File: types/vector2js/vector2js-tests.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+const res2 = v.addSelf(p).mulSelf(p).flipXYSelf().subSelf(o).divSelf(2).rotateDegreesSelf(90).rotateRadiansSelf(3.14);
+
+const res3 = p.add({ x: 1, y: 1 });
+const f1 = Vector2.fromArray([1, 1]);
+const f2 = Vector2.fromDegrees(45);
+const f3 = Vector2.fromRadians(0);
+const f4 = Vector2.fromString('1,1');
+
+const d = p.dot(v);
+const s = p.toString();
```
Comment (peterblazejewicz) [inline types/vector2js/vector2js-tests.ts:18]:
```suggestion
import Vector = require('vector2js');

const o = new Vector(0, 0);
const p = new Vector(1, 1);
const v = new Vector(1, 2);

const res1 = p.add(o).clone().mul(v).flipXY().sub(o).div(2).rotateDegrees(90).rotateRadians(3.14).cross(p);

const res2 = v.addSelf(p).mulSelf(p).flipXYSelf().subSelf(o).divSelf(2).rotateDegreesSelf(90).rotateRadiansSelf(3.14);

const res3 = p.add({ x: 1, y: 1 });
const f1 = Vector.fromArray([1, 1]);
const f2 = Vector.fromDegrees(45);
const f3 = Vector.fromRadians(0);
const f4 = Vector.fromString('1,1');

const d = p.dot(v);
const s = p.toString();
const s2 = p.format('{ %x; %y }');

Vector.zero; // $ExpectType Vector
Vector.one; // $ExpectType Vector
Vector.up; // $ExpectType Vector
Vector.down; // $ExpectType Vector
Vector.left; // $ExpectType Vector
Vector.right; // $ExpectType Vector
Vector.upLeft; // $ExpectType Vector
Vector.downLeft; // $ExpectType Vector
Vector.upRight; // $ExpectType Vector
Vector.downRight; // $ExpectType Vector
```

File: types/vector2js/index.d.ts (hunk 2 above)
Comment (Leo-Nicolle) [reply types/vector2js/index.d.ts]: done

File: types/vector2js/tsconfig.json [hunk 5]
```diff
@@ -0,0 +1,23 @@
+{
+    "compilerOptions": {
+        "module": "commonjs",
+        "lib": [
+            "es5"
```
Comment (peterblazejewicz) [inline types/vector2js/tsconfig.json:5]:
```suggestion
            "es6"
```
the native package is in ES5 clearly, but here we're describing the shape of the JS api, so we can use ES6 features, like class, hence the 'es6' is default on DT

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@Leo-Nicolle thanks!

**Comment by Leo-Nicolle** [comment]
Ready to merge

**Pull request merged**